### ISSUE TITLE
Typescript build error in utils.ts

### ISSUE THREAD

**Issue by zhangela** [markdoc/markdoc#301]
### What happened?

After adding markdoc to my typescript project, running `yarn run build` gives me this error: 

```
./node_modules/@markdoc/markdoc/src/utils.ts:134:3
Type error: Type '({ type: string; start: number; end: number; content: string; } | { type: string; tag: string; attrs: [string, string][] | null; map: [number, number] | null; nesting: 0 | 1 | -1; level: number; ... 9 more ...; end: number; } | { ...; })[]' is not assignable to type 'Token[]'.
  Type '{ type: string; start: number; end: number; content: string; } | { type: string; tag: string; attrs: [string, string][] | null; map: [number, number] | null; nesting: 0 | 1 | -1; level: number; ... 9 more ...; end: number; } | { ...; }' is not assignable to type 'Token'.
    Type '{ type: string; start: number; end: number; content: string; }' is missing the following properties from type 'Token': tag, attrs, map, nesting, and 12 more.

  132 |   });
  133 |
> 134 |   return output;
      |   ^
  135 | }
  136 |
```

### To reproduce

1. Add markdoc to a new next.js project
2. run `yarn run build`

**Comment by adreyfus-stripe** [comment]
Hi @zhangela! Could you provide more details? I tried to reproduce by creating a new nextjs project via `create-next-app@latest` and wasn't able to get the same error. What version of Next.js are you using and did you follow the steps outlined in https://markdoc.dev/docs/nextjs ?

**Comment by zhangela** [comment]
Yep I did follow the steps in  https://markdoc.dev/docs/nextjs. 

To be clear, this is a Typescript error. It shouldn't matter which Typescript version I use. The error is happening because the return type is wrong in `@markdoc/markdoc/src/utils.ts:134`.

**Comment by adreyfus-stripe** [comment]
Thanks for verifying, I was asking about your next.js set up because you shouldn't see blocking ts errors from markdoc if you set it up via the suggested happy path since it's a dependency and should be skipped.  

Digging into this a bit more it does look like the return type is wrong in that util, we'll get a ticket made to patch that, but in the meantime I suggest updating your next/tsconfig to skip linting libraries so you can unblock yourself in the meantime. Thanks for flagging!

**Comment by zhangela** [comment]
I already have `"skipLibCheck": true` in my tsconfig :/
